SonarQube.Client/Requests/PagedRequestBase.cs

```diff
@@ -69,6 +73,11 @@ public async override Task<TResponseItem[]> InvokeAsync(HttpClient httpClient, C
             return allResponseItems.ToArray();
         }
 
+        private bool ReachedMaxPageNumber()
+        {
+            return MaxPageNumber.HasValue && Page > MaxPageNumber.Value;
```

Comment:
Is the problem the maximum number of pages or the maximum number of results?

Do any other APIs have this limitation on the maximum number of results?